indexer/services/ender/src/scripts/dydx_get_order_status.sql

```diff
@@ -1,19 +1,38 @@
 /**
-  Returns the order status given the total filled amount, the order size and whether the order was cancelled.
+  Returns the order status given the total filled amount, the order size, whether the order was
+  cancelled, order flags, and time in force.
+  * The obvious case is if totalFilled >= size, then the order status should always be `FILLED`.
+  * The difficult case is if totalFilled < size after a fill, then we need to keep the following
+  * cases in mind:
+  * - Stateful Orders - All cancelations are onchain events, so the order can be `OPEN` or 
+  *   `BEST_EFFORT_CANCELED` if the order is in the CanceledOrdersCache.
+  * - Shortterm FOK - FOK orders can never be `OPEN`, since they don't rest on the orderbook, so
+  *   totalFilled cannot be < size.
+  * - Shortterm IOC - Protocol guarantees that an IOC order will only ever be filled in a single
+  *   block, so status should be `CANCELED`.
+  * - Shortterm Limit & Postonly - If the order is in the CanceledOrdersCache, then it should be
+  *   set to `BEST_EFFORT_CANCELED`, otherwise `OPEN`.
 */
-CREATE OR REPLACE FUNCTION get_order_status(total_filled numeric, size numeric, is_cancelled boolean)
+CREATE OR REPLACE FUNCTION get_order_status(total_filled numeric, size numeric, is_cancelled boolean, order_flags bigint, time_in_force text)
 RETURNS text AS $$
-DECLARE
-    order_status text;
 BEGIN
-    IF is_cancelled = true THEN
-        order_status = 'BEST_EFFORT_CANCELED';
-    ELSIF total_filled >= size THEN
-        order_status = 'FILLED';
+    IF total_filled >= size THEN
+        RETURN 'FILLED';
+    /** Order flag of 64 is a long term order */
```

Comment:
should we use the same terminology as in the function description? long term -> stateful